tests/_support/AcceptanceTester.php

```diff
@@ -357,20 +357,26 @@ public function optInForRegistration() {
    */
   public function optInForSubscription() {
     $i = $this;
-    $i->waitForElementClickable('#mailpoet_woocommerce_checkout_optin');
-    $i->checkOption('#mailpoet_woocommerce_checkout_optin');
+    $checked = $i->executeJS('return document.getElementById("mailpoet_woocommerce_checkout_optin").checked');
+    if (!$checked) {
+      $i->waitForElementClickable('[data-automation-id="woo-commerce-subscription-opt-in"]');
+      $i->click('[data-automation-id="woo-commerce-subscription-opt-in"]');
+    }
   }
 
   /**
    * Uncheck the option for subscribing to the WC list
    */
   public function optOutOfSubscription() {
     $i = $this;
-    $i->waitForElementClickable('#mailpoet_woocommerce_checkout_optin');
-    $i->uncheckOption('#mailpoet_woocommerce_checkout_optin');
+    $checked = $i->executeJS('return document.getElementById("mailpoet_woocommerce_checkout_optin").checked');
+    if ($checked) {
+      $i->waitForElementClickable('[data-automation-id="woo-commerce-subscription-opt-in"]');
+      $i->click('[data-automation-id="woo-commerce-subscription-opt-in"]');
+    }
```

Comment:
Maybe we should put this into a private method? As you prefer...

Reply:
I'll leave it as is. I think in this case it is slightly better for readability.